### PR TITLE
Support detecting GitLab commit targets in Azure DevOps

### PR DESCRIPTION
**Pull request by aliscott**

### CODE REVIEW

File: src/detect/azureDevOpsPipelines.test.ts [hunk 1]
```diff
@@ -128,12 +136,23 @@ describe('AzureDevOpsPipelinesDetector', () => {
         expect(logs).not.toContain('MY_GITHUB_TOKEN_VALUE');
       });
 
-      it('detects GitHub PR if $SYSTEM_PULLREQUEST_PULLREQUESTNUMBER is set', () => {
+      it('detects GitHub PR if $SYSTEM_PULLREQUEST_PULLREQUESTNUMBER and $BUILD_SOURCEVERSION are set', () => {
         expect(detector.detect()).toEqual(expectedPrResult);
       });
 
-      it('does not detect Github PR if targetType is set to commit', () => {
+      it('does detects Github commit if targetType is set to commit', () => {
```
Comment (vdmgolub) [inline src/detect/azureDevOpsPipelines.test.ts:143]:
```suggestion
      it('detects Github commit if targetType is set to commit', () => {
```

File: src/detect/azureDevOpsPipelines.test.ts [hunk 2]
```diff
@@ -128,12 +136,23 @@ describe('AzureDevOpsPipelinesDetector', () => {
...
         detector = new AzureDevOpsPipelinesDetector({ targetType: 'commit' });
+        expect(detector.detect()).toEqual(expectedCommitResult);
+      });
+
+      it('detects GitLab commit if only $BUILD_SOURCEVERSION is set', () => {
```
Comment (vdmgolub) [inline src/detect/azureDevOpsPipelines.test.ts:148]:
```suggestion
      it('detects GitHub commit if only $BUILD_SOURCEVERSION is set', () => {
```

File: src/detect/azureDevOpsPipelines.ts [hunk 3]
```diff
@@ -89,6 +89,11 @@ export class AzureDevOpsPipelinesDetector extends BaseDetector {
       }
     }
 
+    if (!targetRef && this.shouldDetectTargetType('commit')) {
```
Comment (vdmgolub) [inline src/detect/azureDevOpsPipelines.ts:92]: Seems like `!targetRef` is redundant, but doesn't hurt to keep :)
```suggestion
    if (this.shouldDetectTargetType('commit')) {
```

**Review by vdmgolub — APPROVED** [review]
Looks good, couple comments with typo fixes.

File: src/detect/azureDevOpsPipelines.ts (hunk 3 above)
Comment (aliscott) [reply src/detect/azureDevOpsPipelines.ts:92]: I think we need it. Otherwise it overwrites the targetRef that is found when looking for a PR environment.

Comment (vdmgolub) [reply src/detect/azureDevOpsPipelines.ts:92]: Each if statement excludes the other, so if `this.shouldDetectTargetType('commit')` is true, `this.shouldDetectTargetType('pull-request') || this.shouldDetectTargetType('merge-request')` will always be false, right? I may be missing something 🤔 . Anyway, it's okay to keep :)

Comment (aliscott) [reply src/detect/azureDevOpsPipelines.ts:92]: By default compost detects both PRs and commits unless it's limited it with the `--target-type` flag.  So both can be true if there's no flag to limit the detection.

Comment (vdmgolub) [reply src/detect/azureDevOpsPipelines.ts:92]: Ah, got it! Thank you for the explanation 👍!

**Pull request merged**